rclpy/src/rclpy_common/src/common.c

```diff
@@ -34,63 +59,53 @@ rclpy_convert_to_py_qos_policy(void * profile)
     return NULL;
   }
 
+  // start qos setting
   rmw_qos_profile_t * qos_profile = (rmw_qos_profile_t *)profile;
+  rclpy_qos_profile_t pyqos;
+  init_rclpy_qos_profile(&pyqos);
+  int set_result = -1;
 
-  PyObject * pyqos_depth = PyLong_FromSize_t(qos_profile->depth);
-  if (!pyqos_depth) {
-    Py_DECREF(pyqos_profile);
```

Comment:
We should still decref `pyqos_profile` if there is an error between here and any return.

Reply:
Yeah good catch - updating